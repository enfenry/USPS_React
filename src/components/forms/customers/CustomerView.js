import React from 'react';
import { Button } from 'reactstrap';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { displayById }  from '../formUtils';

const CustomerView = props => {
    const { handleHide, initialValues, addresses } = props;

    return (
        <div>
            <div>
                <div>
                    <span>First Name: </span>
                    <p>{initialValues.firstname}</p>
                </div>
            </div>
            <div>
                <div>
                    <span>Last Name: </span>
                    <p>{initialValues.lastname}</p>
                </div>
            </div>
            <div>
                <div>
                    <span>Email: </span>
                    <p>{initialValues.emailaddress1 || 'None'}</p>
                </div>
            </div>

            <div>
                <div>
                    <span>Address Name: </span>
                    <p>{displayById(addresses, "ss_customaddressid", initialValues._ss_contactcustomaddress_value, "ss_name")}</p>
                </div>
            </div>
            <div className="control">
                <Button color="secondary" onClick={handleHide}>Cancel</Button>
            </div>
        </div>
    )
}

CustomerView.propTypes = {
    handleHide: PropTypes.func,
    name: PropTypes.string,
    initialValues: PropTypes.object,
    addresses: PropTypes.arrayOf(PropTypes.object),
    children: PropTypes.oneOfType([
        PropTypes.arrayOf(PropTypes.node),
        PropTypes.node
    ])
}

function mapStateToProps(state) {
    return {
        addresses: state.addressesReducer.addresses
    }
}

export default connect(
    mapStateToProps
)(CustomerView);