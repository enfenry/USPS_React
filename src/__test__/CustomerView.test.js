import React from 'react';
import CustomerView from '../components/forms/customers/CustomerView';
import renderer from 'react-test-renderer';

test('CustomerView renders correctly', () => {

    const component = renderer.create(
        <CustomerView
            initialValues={{
                firstname: "example first name",
                lastname: "example last name",
                emailaddress1: "example email address",
            }}
        />
    );

    let tree = component.toJSON();

    expect(tree).toMatchSnapshot();
});