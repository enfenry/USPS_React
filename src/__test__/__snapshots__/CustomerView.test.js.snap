// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerView renders correctly 1`] = `
<div>
  <div>
    <div>
      <span>
        First Name: 
      </span>
      <p>
        example first name
      </p>
    </div>
  </div>
  <div>
    <div>
      <span>
        Last Name: 
      </span>
      <p>
        example last name
      </p>
    </div>
  </div>
  <div>
    <div>
      <span>
        Email: 
      </span>
      <p>
        example email address
      </p>
    </div>
  </div>
  <div
    className="control"
  >
    <button
      aria-label={null}
      className="btn btn-secondary"
      onClick={[Function]}
    >
      Cancel
    </button>
  </div>
</div>
`;
